extensions-core/datasketches/src/main/java/io/druid/query/aggregation/datasketches/tuple/ArrayOfDoublesSketchSetOpPostAggregator.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.query.aggregation.datasketches.tuple;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.yahoo.sketches.Util;
+import com.yahoo.sketches.tuple.ArrayOfDoublesSketch;
+
+import io.druid.java.util.common.IAE;
+import io.druid.query.aggregation.AggregatorUtil;
+import io.druid.query.aggregation.PostAggregator;
+
+public class ArrayOfDoublesSketchSetOpPostAggregator extends ArrayOfDoublesSketchMultiPostAggregator
+{
+
+  private final ArrayOfDoublesSketchOperations.Func func;
+  private final int nominalEntries;
+  private final int numberOfValues;
+
+  @JsonCreator
+  public ArrayOfDoublesSketchSetOpPostAggregator(
+      @JsonProperty("name") final String name,
+      @JsonProperty("func") final String func,
+      @JsonProperty("nominalEntries") final Integer nominalEntries,
+      @JsonProperty("numberOfValues") final Integer numberOfValues,
+      @JsonProperty("fields") List<PostAggregator> fields
+  )
+  {
+    super(name, fields);
+    this.func = ArrayOfDoublesSketchOperations.Func.valueOf(func);
```

Comment:
Instead of using Enum enumeration, could this functionality be implemented as three separate subclasses of `ArrayOfDoublesSketchSetOpPostAggregator`, i. e. three different aggregators?

Reply:
Perhaps. There is always more than one way. How important is this?

Reply:
Now I also think that the current was is preferable, because composition is better than subclassing.